domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -884,63 +884,66 @@ public void saveBatch(String table, JsonArray entities, Handler<AsyncResult<Resu
     saveBatch(table, entities, DEFAULT_JSONB_FIELD_NAME, replyHandler);
   }
 
-  public void saveBatch(String table, JsonArray entities, String column, Handler<AsyncResult<ResultSet>> replyHandler) {
+  public void saveBatch(String table, JsonArray entities, String column,
+    Handler<AsyncResult<ResultSet>> replyHandler) {
+
+    client.getConnection(res -> {
+      if (res.failed()) {
+        replyHandler.handle(Future.failedFuture(res.cause()));
+        return;
+      }
+      saveBatch(res, table, entities, column, x -> {
+        res.result().close();
+        replyHandler.handle(x);
+      });
+    });
+  }
+
+  private void saveBatch(AsyncResult<SQLConnection> sqlConnection, String table,
+    JsonArray entities, String column, Handler<AsyncResult<ResultSet>> replyHandler) {
+
     long start = System.nanoTime();
     if (entities == null || entities.isEmpty()) {
       // return empty result
       ResultSet resultSet = new ResultSet(
-          Collections.singletonList(idField), Collections.emptyList(), null);
+        Collections.singletonList(idField), Collections.emptyList(), null);
       replyHandler.handle(Future.succeededFuture(resultSet));
       return;
     }
-    client.getConnection(res -> {
-      if (res.failed()) {
-        replyHandler.handle(Future.failedFuture(res.cause()));
+    if (log.isInfoEnabled()) {
+      log.info("starting: saveBatch size=" + entities.size());
+    }
+    StringBuilder sql = new StringBuilder()
+      .append(INSERT_CLAUSE)
+      .append(schemaName).append(DOT).append(table)
+      .append(" (").append(column).append(") VALUES (?)");
+    for (int i = 1; i < entities.size(); i++) {
+      sql.append(",(?)");
+    }
+    sql.append(" RETURNING ").append(idField);
+
+    SQLConnection connection = sqlConnection.result();
```

Comment:
Defensive programming: replyHandler needs to be called with a failedFuture if  sqlConnection==null or sqlConnection.failed().